packages/ui/src/working-groups/hooks/useOpenings.ts

```diff
@@ -1,17 +1,34 @@
+import { useMemo } from 'react'
+
 import { useGetWorkingGroupOpeningsQuery } from '../queries'
-import { asWorkingGroupOpening } from '../types'
+import { asWorkingGroupOpening, WorkingGroupOpening } from '../types'
 
 interface UseOpeningsParams {
   groupId?: string
+  type?: 'open' | 'past'
 }
 
-export const useOpenings = ({ groupId }: UseOpeningsParams = {}) => {
+export const useOpenings = ({ groupId, type }: UseOpeningsParams = {}) => {
   const { loading, data } = useGetWorkingGroupOpeningsQuery({ variables: { group_eq: groupId } })
 
+  const typeFilter: (opening: WorkingGroupOpening) => boolean = useMemo(() => {
+    switch (type) {
+      case 'open':
+        return (opening) => isOpeningOpen(opening)
+      case 'past':
+        return (opening) => !isOpeningOpen(opening)
+      default:
+        return () => true
+    }
+  }, [type])
```

Comment:
This looks complex:
- there's `useMemo()` where it rather should be `useCallback()`
- because of the optionality of the `type` you had to use a switch where `default` is not used now.

```suggestion
  const typeFilter: getTypeFilter( type )
```

where:

```ts
const getTypeFilter = (type: 'open' | 'past') => {
  if (type === 'open' ) {
    return isOpeningOpen
  }
  return (opening: WorkingGroupOpening) => !isOpeningOpen(opening)
}
```

Reply:
And there the `isOpeningOpen` could be optionally inlined there.